src/main/kotlin/com/expedia/graphql/schema/schemaAnnotations.kt

```diff
@@ -11,31 +10,49 @@ import graphql.schema.GraphQLDirective
 import graphql.schema.GraphQLInputType
 import kotlin.reflect.KAnnotatedElement
 import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
 import kotlin.reflect.KParameter
+import kotlin.reflect.KProperty1
 import kotlin.reflect.KType
 import kotlin.reflect.full.declaredMemberFunctions
 import kotlin.reflect.full.findAnnotation
+import kotlin.reflect.full.memberProperties
 import com.expedia.graphql.annotations.GraphQLDirective as DirectiveAnnotation
 
 internal fun KAnnotatedElement.graphQLDescription(): String? {
-    var prefix = this.getDeprecationReason()?.let { "DEPRECATED" }
-    if (null == prefix && this.isGraphQLExperimental()) {
-        prefix = "EXPERIMENTAL"
-    }
+    val directiveNames = listOfDirectives()
 
     val description = this.findAnnotation<GraphQLDescription>()?.value
 
-    return if (null != description) {
-        if (null != prefix) {
-            "$prefix: $description"
-        } else {
-            description
+    return when {
+        description != null && directiveNames.isNotEmpty() -> {
+            """$description
+            |
+            |Directives: ${directiveNames.joinToString(", ")}
+            """.trimMargin()
+            }
+        description == null && directiveNames.isNotEmpty() -> {
+            "Directives: ${directiveNames.joinToString(", ")}"
         }
-    } else {
-        prefix
+        else -> description
     }
 }
 
+private fun KAnnotatedElement.listOfDirectives(): List<String> {
+    val directiveNames = mutableListOf(this.getDeprecationReason()?.let { "DEPRECATED" })
+
+    directiveNames.addAll(this.annotations
+            .filter { it.getDirectiveInfo() != null }
+            .map {
+                when {
+                    it.getDirectiveInfo()?.name != "" -> "@${it.getDirectiveInfo()?.name}"
+                    else -> "@${it.annotationClass.simpleName}"
```

Comment:
are directives generally lower case? or just start with lower case? or it could be anything?